### PR TITLE
[core] Add codemod to move l10n translation

### PR DESCRIPTION
**Pull request by alexfauquette**
Part of #6997

### CODE REVIEW

File: packages/x-codemod/src/v6.0.0/text-props-to-localeText/expected-expression-values.spec.js [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+      secondsClockNumberText: custom_getSecondsClockNumberText,
+      calendarViewSwitchingButtonAriaLabel: custom_getViewSwitchingButtonText,
+      openPreviousView: custom_leftArrowButtonText,
+      previousMonth: custom_leftArrowButtonText,
+      openNextView: custom_rightArrowButtonText,
+      nextMonth: custom_rightArrowButtonText
+    }} />
+
+  {/* Stand alone */}
+  <CalendarPicker
```
Comment (flaviendelangle) [inline packages/x-codemod/src/v6.0.0/text-props-to-localeText/expected-expression-values.spec.js:243]: I went to fast...
Those two components don't have a `localeText` prop.
We would need to add a `LocalizationProvider` around them.
Or to add support for `localeText`

File: packages/x-codemod/src/v6.0.0/text-props-to-localeText/index.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  'MobileTimePicker',
+  'StaticDatePicker',
+  'StaticDateRangePicker',
+  'StaticDateTimePicker',
+  'StaticTimePicker',
+  'TimePicker',
+];
+
+/**
+ * @param {import('jscodeshift').FileInfo} file
```
Comment (LukasTy) [inline packages/x-codemod/src/v6.0.0/text-props-to-localeText/index.js:69]: Did you experience problems authoring a ts file? 🤔

File: packages/x-codemod/src/v6.0.0/text-props-to-localeText/text-props-to-localeText.test.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import jscodeshift from 'jscodeshift';
+import transform from './index';
+import readFile from '../../util/readFile';
+
+function read(fileName) {
+  return readFile(path.join(__dirname, fileName));
+}
+
+describe('@mui/codemod', () => {
+  describe('v5.0.0', () => {
```
Comment (LukasTy) [inline packages/x-codemod/src/v6.0.0/text-props-to-localeText/text-props-to-localeText.test.js:12]: Could you take a look at how the existing tests are described? 🤔

**Review by LukasTy — COMMENTED** [review]
Have you looked at "The process of adding a new codemod" described in https://github.com/mui/mui-x/issues/6997

It seems that this codemod should be pretty safe, isn't it?
To follow the process of mui-material I've suggested to also add such codemods to the `preset-safe`, ideally update it's tests and also update the readme. 🤔 
What do you think?

File: packages/x-codemod/README.md [hunk 4]
```diff
@@ -91,4 +92,37 @@ Renames `locale` into `adapterLocale` (or `LocalizationProvider`)
...
+ <DatePicker
+-  cancelText="Cancel"
++  localeText={{
++    cancelButtonLabel: "Cancel"
++  }}
+ />
+```
+
+```sh
+npx @mui/x-codemod v6.0.0/localization-provider-rename-locale <path>
```
Comment (LukasTy) [inline packages/x-codemod/README.md:109]:
```suggestion
npx @mui/x-codemod v6.0.0/text-props-to-localeText <path>
```

File: packages/x-codemod/README.md [hunk 5]
```diff
@@ -91,4 +92,37 @@ Renames `locale` into `adapterLocale` (or `LocalizationProvider`)
 npx @mui/x-codemod v6.0.0/localization-provider-rename-locale <path>
 ```
 
+#### `text-props-to-localeText`
+
+Replace props used for translation such as `cancelText` to their corresponding `localeText` key.
```
Comment (LukasTy) [inline packages/x-codemod/README.md:97]:
```suggestion
Replace props used for localization such as `cancelText` to their corresponding `localeText` key.
```

File: packages/x-codemod/README.md [hunk 6]
```diff
@@ -91,4 +92,37 @@ Renames `locale` into `adapterLocale` (or `LocalizationProvider`)
...
++    cancelButtonLabel: "Cancel"
++  }}
+ />
+```
+
+```sh
+npx @mui/x-codemod v6.0.0/localization-provider-rename-locale <path>
+```
+
+If you were using always the same text value in all your components, consider moving those translation from the component to the `LocalizationProvider` by hand.
```
Comment (LukasTy) [inline packages/x-codemod/README.md:112]:
```suggestion
If you were always using the same text value in all your components, consider moving those translation from the component to the `LocalizationProvider` by hand.
```

File: packages/x-codemod/README.md [hunk 7]
```diff
@@ -91,4 +92,37 @@ Renames `locale` into `adapterLocale` (or `LocalizationProvider`)
...
+```diff
+<LocalizationProvider
+  dateAdapter={AdapterDayjs}
++ localeText={{ cancelButtonLabel: "Cancel" }}
+>
+ <DatePicker
+-  localeText={{ cancelButtonLabel: "Cancel" }}
+ />
+ <DateTimePicker
+-  localeText={{ cancelButtonLabel: "Cancel" }}
```
Comment (LukasTy) [inline packages/x-codemod/README.md:123]: Maybe having something different than the default makes a bit more sense in this case? 🤔 
```suggestion
+ localeText={{ cancelButtonLabel: "Cancelar" }}
>
 <DatePicker
-  localeText={{ cancelButtonLabel: "Cancelar" }}
 />
 <DateTimePicker
-  localeText={{ cancelButtonLabel: "Cancelar" }}
```

**Review by LukasTy — APPROVED** [review]
Great job! 🙌 💯

**Review by flaviendelangle — CHANGES_REQUESTED** [review]
`ClockPicker` and `CalendarPicker` don't have a `localeText` prop

**Comment by alexfauquette** [comment]
> ClockPicker and CalendarPicker don't have a localeText prop

Let me realize we could have aproblem in terms of order to apply codemod: component renaming

`ClockPicker` is renamed `TimeClock`. So this codemod should be applied before renaming components

**Comment by LukasTy** [comment]
> `ClockPicker` is renamed `TimeClock`. So this codemod should be applied before renaming components

Good point. 👌 
I'd encourage adding this case to `preset-safe.spec` so that we can adjust once the follow-up codemod for renaming is added 😉

File: packages/x-codemod/tsconfig.json [hunk 8]
```diff
@@ -3,5 +3,10 @@
   "compilerOptions": {
     "types": ["react", "mocha", "node"]
   },
-  "include": ["src/**/*", "../../node_modules/@mui/monorepo/test/utils/initMatchers.ts"]
+  "include": [
+    "src/**/*",
+    "../../node_modules/@mui/monorepo/test/utils/initMatchers.ts",
+    "../../node_modules/@mui/material/themeCssVarsAugmentation",
+    "../../node_modules/date-fns"
```
Comment (alexfauquette) [inline packages/x-codemod/tsconfig.json:10]: I don't know why the typescript on `@mui/x-codemod` complains that `Locale` is not defined in `useMaskedInput` whereas typescript does not complain on `@mui/x-date-icker` itself

So I added is in the tsconfig, but I assume their is a smarter wai of doing that
```
    "../../node_modules/date-fns"
```

Comment (LukasTy) [reply packages/x-codemod/tsconfig.json:10]: Hmm... Have you tried adding `"skipLibCheck": true,`? That's one thing that is different between these projects. 🤔

Comment (alexfauquette) [reply packages/x-codemod/tsconfig.json:10]: Yes, but does not solve the problem

**Review by flaviendelangle — APPROVED** [review]
Great job !

**Pull request merged**